support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -197,115 +191,146 @@ function getServerSideParticipations(): Participations | null | undefined {
+	const buildParticipation = (
+		test: AmountsTest,
+		testName: string,
+		variantName: string,
+	): Participations | undefined => {
+		// Check if we actually want to track this test
+		const pathMatches = targetPageMatches(
+			path,
+			'/??/contribute|thankyou(/.*)?$',
+		);
+		if (pathMatches && test.variants.length > 1 && test.isLive) {
+			return {
+				[testName]: variantName,
+			};
+		}
+	};
+
+	// Is an amounts test defined in the url?
 	const urlTest = getAmountsTestFromURL(acquisitionDataTests);
-	if (urlTest != null) {
-		return urlTest;
+	if (urlTest) {
+		// Attempt to find urlTest in the configured amounts tests
+		const candidate = amounts.find((t) => {
+			if (t.isLive) {
+				return t.liveTestName === urlTest.name;
+			} else {
+				return t.testName === urlTest.name;
+			}
+		});
+		if (candidate) {
+			const variants = candidate.variants;
+			if (variants.length) {
+				const variant =
+					variants.find((variant) => variant.variantName === urlTest.variant) ??
+					variants[0];
+				const amountsParticipation = buildParticipation(
+					candidate,
+					urlTest.name,
+					variant.variantName,
+				);
+				return {
+					selectedAmountsVariant: {
+						...variant,
+						testName: urlTest.name,
+					},
+					amountsParticipation,
+				};
+			}
+		}
 	}
 
-	if (!settings.amounts) {
-		return null;
-	}
+	// No url test was found, use targeting
+	let targetedTest: AmountsTest | undefined;
 
-	const amounts = settings.amounts;
-
-	/*
-	An amounts test can be in one of two forms:
-
-	Country test:
-	  Bespoke tests targeted at one or more geographical countries
-	  `targeting` object will include a `countries` attribute
-	    - a String array containing 2-letter ISO country codes
-	  When the `isLive` boolean is `false`:
-	    - the test is ignored; users will see their appropriate region test
-	  When the `isLive` boolean is `true`:
-	    - users will be randomly segregated into an AB test and see the appropriate variant
-	    - analytics will use the `liveTestName` label, if available, else the `testName` label
-	  A country can appear in more than one country test:
-	    - if 2+ live tests include the country, the test with the lowest `order` value will display
-
-	Region test:
-	  Evergreen tests, one per geographical region
-	  `targeting` object will include a `region` attribute
-	    - the region label, as defined by the Region type
-	  When the `isLive` boolean is `false`:
-	    - the CONTROL variant will display
-	    - analytics will use the `testName` label
-	  When the `isLive` boolean is `true`:
-	    - users will be randomly segregated into an AB test and see the appropriate variant
-	    - analytics will use the `liveTestName` label
-	*/
-	const targetTestArray = amounts.filter(
-		(t) =>
-			t.isLive &&
-			t.targeting.targetingType === 'Country' &&
-			t.targeting.countries.includes(country),
-	);
-	let targetTest;
+	// First try country-targeted tests
 	const source = getSourceFromAcquisitionData() ?? '';
```

Comment:
I've just pushed after a rebase that brings in the change to exclude AMP/AN from country-targeted tests